src/transformer_deploy/utils/token_classifier.py

```diff
@@ -34,68 +34,70 @@
+            tensor: torch.Tensor = torch.from_dlpack(output.to_dlpack())
+            return TokenClassifierOutput(logits=tensor)
 
 
 class TritonPythonModel:
     tokenizer: PreTrainedTokenizer
     device: str
-    # change aggregation strategy here
-    aggregation_strategy = AggregationStrategy.FIRST
-    ignore_labels = ["O"]
+    model: BertForTokenClassification
 
     def initialize(self, args: Dict[str, str]) -> None:
         """
         Initialize the tokenization process
         :param args: arguments from Triton config file
         """
-        current_path: str = os.path.join(args["model_repository"], args["model_version"])
+        current_name: str = os.path.join(args["model_repository"], args["model_version"])
         target_model = args["model_name"].replace("_inference", "_model")
         self.device = "cpu" if args["model_instance_kind"] == "CPU" else "cuda"
-
-        def inference_triton(
-            input_ids: torch.Tensor,
-            token_type_ids: torch.Tensor,
-            attention_mask: torch.Tensor,
-        ) -> np.ndarray:
-            input_ids = input_ids.type(dtype=torch.int32)
-            token_type_ids = token_type_ids.type(dtype=torch.int32)
-            attention_mask = attention_mask.type(dtype=torch.int32)
-            inputs = [
-                pb_utils.Tensor.from_dlpack("input_ids", torch.to_dlpack(input_ids)),
-                pb_utils.Tensor.from_dlpack("token_type_ids", torch.to_dlpack(token_type_ids)),
-                pb_utils.Tensor.from_dlpack("attention_mask", torch.to_dlpack(attention_mask)),
-            ]
-            inference_request = pb_utils.InferenceRequest(
-                model_name=target_model,
-                requested_output_names=["output"],
-                inputs=inputs,
-            )
-            inference_response = inference_request.exec()
-            if inference_response.has_error():
-                raise pb_utils.TritonModelException(inference_response.error().message())
-            else:
-                output = pb_utils.get_output_tensor_by_name(inference_response, "output")
-                tensor: torch.Tensor = torch.from_dlpack(output.to_dlpack())
-                return tensor.detach().cpu().numpy()
-
-        self.model = inference_triton
-        self.tokenizer = AutoTokenizer.from_pretrained(current_path)
-        self.config = AutoConfig.from_pretrained(current_path)
+        self.model = BertForTokenClassification(model_name=current_name, model_path=target_model)
+        self.model.to(device=self.device)
+        self.tokenizer = AutoTokenizer.from_pretrained(current_name)
+        # self.config = AutoConfig.from_pretrained(current_name)
+        self.pipeline = TokenClassificationPipeline(
+            task="token-classification",
+            model=self.model,
+            tokenizer=self.tokenizer,
+            aggregation_strategy=AggregationStrategy.SIMPLE,
```

Comment:
I don't know how much it is important, but if it is the case we can pass it as argument maybe (the aggregation strategy)

Reply:
I don't think that's something you would change between 2 calls, that's why it seems Ok to be hardcoded. In any case, there are plenty of parts you may want to customize, the idea it to get a basic working code.

Reply:
so no need to customize it :100: